### ISSUE TITLE
Can I use EncryptPad with keys from GnuPG?

### ISSUE THREAD

**Issue by Trollwut** [evpo/EncryptPad#65]
Hi folks!

I already have text files encrypted with my GnuPG keys. Now when I want to edit them, I have to decrypt them, open with text editor, save and encrypt again.

Am I able to open them with EncryptPad, so that it fetches the proper private key to de- and encrypt from my local GnuPG key files?

If so, I haven't found out how. :x

**Comment by evpo** [comment]
No, you can't do it at the moment because GnuPG keys are asymmetric (public/private). EncryptPad key is symmetric ([read more...](http://evpo.net/encryptpad/#key-file))

People ask the question a lot. Since this is a community driven project, I think we should implement this feature. It will also improve compatibility with other editors such as [gnupg.vim](http://www.vim.org/scripts/script.php?script_id=3645).

Related issues: [#8](https://github.com/evpo/EncryptPad/issues/8)

**Comment by BjornFJohansson** [comment]
This would be a very nice feature indeed! I stumbled here searching for something to solve a very specific use case I have.

I work as a university teacher and I distribute tests for my students in the form of plain text files.
I usually distribute these ahead of time as encrypted zip or 7z files. I give the students a passphrase to open the files at the start of the test. This way I do not have to worry about students not being able to download files due to flaky wifi at university.

Students differ in computer savviness however and some have issues using the several different software needed and understanding what a zip file is etc. On the other scale I have students that have managed to break the encryption of the zip files I used in the beginning:(

Encryptpad seems very easy to use and would probably work very well. It would be perfect if I could use a my private key for encryption and distribute the public key with the exam files.

Is this feature on the horizon?

**Comment by evpo** [comment]
It's an interesting use case.

To answer your question, the feature is not simple. It may take some time to implement. There will be a lot of back-end and UI changes to support it.

You could probably use a temporary EncryptPad symmetric key for now if you don't want to work with a passphrase.

**Comment by PowerPress** [comment]
Please add support for asymmetric encryption. GPG really needs a good clean safe and easy editor to work with encrypting/decrypting data.
